util/build_configs/cray-internal/generate-modulefile.bash

```diff
@@ -135,9 +139,11 @@ if { [string match aarch64 $CHPL_HOST_ARCH] } {
 
 if { ! [ info exists env(PE_ENV) ] } {
     module load PrgEnv-gnu
+    set compiler GNU
+} else {
+    set compiler $env(PE_ENV)
```

Comment:
I can't remember exactly why this was fixing the issue. i.e. loading a PrgEnv module should set PE_ENV, but that was going wrong with lmod probably?

In any case, I remember that this was helping with one of the issues we had.

Reply:
See https://lmod.readthedocs.io/en/latest/095_tcl2lua.html for the explanation. Effectively `set compiler $env(PE_ENV)` gets evaluated before the `module load PrgEnv-gnu` actually runs